pkg/api/handlers/libpod/generate.go

```diff
@@ -43,12 +42,16 @@ func GenerateSystemd(w http.ResponseWriter, r *http.Request) {
 		Name:            query.Name,
 		New:             query.New,
 		NoHeader:        query.NoHeader,
-		RestartPolicy:   query.RestartPolicy,
 		StopTimeout:     &query.StopTimeout,
 		ContainerPrefix: query.ContainerPrefix,
 		PodPrefix:       query.PodPrefix,
 		Separator:       query.Separator,
 	}
+
+	if _, found := r.URL.Query()["restartpolicy"]; found {
```

Comment:
The parameter is documented as `restartPolicy`, I know that the decoder works because it is case insensitive but we have to support `restartPolicy` as well since this is what is documented and what currently works. Can you just change the `query.RestartPolicy` field to a pointer?

Reply:
Or if you set the default value in the query struct to DefaultRestartPolicy it should also work.

Reply:
Excellent idea